### PR TITLE
Harden code around disabled apps

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
With this change, the app store offers two methods to list apps:
- `List() []apps.App`
- `ListIncludeDisabled() []apps.App `

This way it more clear which kind of apps are returned to the proxy layer.

#### Ticket Link
https://github.com/mattermost/mattermost-plugin-apps/pull/409#issuecomment-1349692548

### CODE REVIEW

File: server/proxy/notify.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func (p *Proxy) notifyUserChannel(channelID, userID string, joined bool, method
 
 	// If the user is a bot, process SubjectBot...Channel; only notify the
 	// app with the matching BotUserID.
-	allApps := p.store.App.AsMap()
+	allApps := p.store.App.ListAsMap()
```
Comment (hanzei) [inline server/proxy/notify.go:77]: This is a behaviour change, and a wanted one I think. @levb Can you please double check?

File: server/store/apps.go [hunk 2]
```diff
@@ -116,34 +118,62 @@ func (s *appStore) Get(appID apps.AppID) (*apps.App, error) {
...
 }
 
-func (s *appStore) AsMap() map[apps.AppID]apps.App {
+func (s *appStore) List() []apps.App {
+	var out []apps.App
+	for _, app := range s.ListAsMap() {
+		out = append(out, app)
+	}
+
+	return out
```
Comment (levb) [inline server/store/apps.go:127]: 1/5 sort them by appID to make predictable?

File: server/store/apps.go [hunk 3]
```diff
@@ -116,34 +118,62 @@ func (s *appStore) Get(appID apps.AppID) (*apps.App, error) {
...
+		if !app.Disabled {
+			out[appID] = app
+		}
 	}
+
 	return out
 }
 
-func SortApps(appsMap map[apps.AppID]apps.App) []apps.App {
+func (s *appStore) ListIncludeDisabled() []apps.App {
```
Comment (levb) [inline server/store/apps.go:151]: 1/5 `s/ListIncludeDisabled/RawList`, less Java?

File: server/store/apps.go [hunk 4]
```diff
@@ -116,34 +118,62 @@ func (s *appStore) Get(appID apps.AppID) (*apps.App, error) {
...
+func (s *appStore) List() []apps.App {
+	var out []apps.App
+	for _, app := range s.ListAsMap() {
+		out = append(out, app)
+	}
+
+	return out
+}
+
+func (s *appStore) ListAsMap() map[apps.AppID]apps.App {
```
Comment (levb) [inline server/store/apps.go:130]: What was wrong with, `AsMap`, that it was not "raw"? 0/5 perhaps, `ListEnabledApps`, and `MapEnabledApps` (and `RawList`, see below?)

**Review by levb — CHANGES_REQUESTED** [review]
nits/naming

File: server/proxy/list.go [hunk 5]
```diff
@@ -56,7 +57,7 @@ func (p *Proxy) GetInstalledApp(appID apps.AppID, checkEnabled bool) (*apps.App,
 }
 
 func (p *Proxy) PingInstalledApps(ctx context.Context) (installed []apps.App, reachable map[apps.AppID]bool) {
-	all := p.store.App.AsMap()
+	all := p.store.App.ListIncludeDisabled()
```
Comment (mickmister) [inline server/proxy/list.go:60]: Why do we want to ping them if they're disabled?

**Review by mickmister — APPROVED** [review]

File: server/proxy/list.go (hunk 5 above)
Comment (levb) [reply server/proxy/list.go:60]: I think it's used in `apps list` to show their online status; 0/5 kinda makes sense.

**Review by levb — APPROVED** [review]

**Pull request merged**